src/library/scala/collection/concurrent/Map.scala

```diff
@@ -102,4 +102,27 @@ trait Map[A, B] extends scala.collection.mutable.Map[A, B] {
       }
   }
 
+  /**
+   * Attempts to compute a mapping for the specified key and its current optionally-mapped value
+   * (`Some` if there is current mapping, `None` if not).
+   * If the function returns `None`, the mapping is removed (or remains absent if initially absent).
+   * If the function itself throws an exception, the exception is rethrown, and the current mapping is left unchanged.
+   *
+   * @param key the key value
+   * @param remappingFunction the function to compute a new mapping
+   * @tparam V1 the type of newly computed value
+   * @return the new value associated with the specified key
+   */
+  override def compute(key: A)(remappingFunction: (A, Option[B]) => Option[B]): Option[B] = {
```

Comment:
`@tailrec`?

Reply:
To annotate with `@trailrec`, `final` modifier is required.
Compiler raises `could not optimize @tailrec annotated method compute: it is neither private nor final so can be overridden`.

Reply:
You can extract an inner tail-recursive function `computeAux`, and end the method with a call to it.